src/core/common/linked_list.hpp

```diff
@@ -409,13 +453,17 @@ template <typename Type> class LinkedList
      * @returns A pointer to the matching entry if one is found, or nullptr if no matching entry was found.
      *
      */
-    template <typename Inidcator> const Type *FindMatching(const Inidcator &aIndicator, const Type *&aPrevEntry) const
+    template <typename Inidcator>
+    const Type *FindMatching(const Type *     aBegin,
+                             const Type *     aEnd,
+                             const Inidcator &aIndicator,
+                             const Type *&    aPrevEntry) const
```

Comment:
Thanks for the two approaches, both of which should work. 
We have thought about `Approach 1` but decided to add a ranged `FindMatching` method to `LinkedList`:
- It's a valid use case to search in a given range of linked list. The method might also be useful for other purposes. 
- It does not increase any code size  if ranged `FindMatching` is not used. (from the size-report we can find that the overall code size actually decreased by 16B)

For `approach 2`, I think it will increase code size so it does not seem to be the best choice. 

Thoughts?

Reply:
SGTM. Thanks.